### PR TITLE
Batch update to plugin

### PR DESCRIPTION
**Pull request by cstamas**
Update parent POM, dependencies and
sort out resolver-util.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -369,6 +389,16 @@ under the License.
...
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-compiler-plugin</artifactId>
+        <configuration>
+          <proc>none</proc>
+          <compilerArgs>
+            <compilerArg>-Xlint:deprecation</compilerArg>
+          </compilerArgs>
```
Comment (slawekjaranowski) [inline pom.xml:399]: We have a param `showDeprecation` for m-compiler-p

File: pom.xml [hunk 2]
```diff
@@ -62,12 +62,13 @@ under the License.
 
   <properties>
     <javaVersion>8</javaVersion>
-    <mavenVersion>3.2.5</mavenVersion>
+    <mavenMinVersion>3.2.5</mavenMinVersion>
+    <mavenVersion>3.9.6</mavenVersion>
```
Comment (slawekjaranowski) [inline pom.xml:66]: I'm not still sure about such way.

We open way to use newer API/methods in plugin which can be only detected in runtime.

I'm for switch to 3.6.3 as minimal and in dependency

Comment (michael-o) [reply pom.xml:66]: I wouldn't do this either in this release.

File: pom.xml [hunk 3]
```diff
@@ -145,6 +146,19 @@ under the License.
       <version>${mavenVersion}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>org.apache.maven.resolver</groupId>
+      <artifactId>maven-resolver-api</artifactId>
+      <version>1.9.18</version>
+      <scope>provided</scope>
```
Comment (michael-o) [inline pom.xml:153]: This does not correspond to the approach we have taken in other plugins: Use the same Resolver API version as with the base Maven Core version.

File: pom.xml [hunk 4]
```diff
@@ -169,6 +183,12 @@ under the License.
       <groupId>org.apache.maven.doxia</groupId>
       <artifactId>doxia-sink-api</artifactId>
       <version>1.12.0</version>
+      <exclusions>
+        <exclusion>
+          <groupId>org.codehaus.plexus</groupId>
+          <artifactId>plexus-container-default</artifactId>
+        </exclusion>
+      </exclusions>
```
Comment (michael-o) [inline pom.xml:191]: Fair enough, will be solved with Doxia 2.0.0.

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by michael-o** [comment]
@cstamas Is this one still required?

**Pull request closed without merging**